src/main/java/io/anserini/ltr/BaseFeatureExtractor.java

```diff
@@ -123,7 +123,7 @@ public BaseFeatureExtractor(IndexReader reader, Qrels qrels, Map<K, Map<String,S
             q = parseQuery(queryText);
             List<String> queryTokens = AnalyzerUtils.tokenize(queryAnalyzer, queryText);
             // Construct the reranker context
-            RerankerContext context = new RerankerContext(searcher, q,
+            RerankerContext<String> context = new RerankerContext<>(searcher, q,
```

Comment:
I think we need `<K>` here instead of `<String>`? Otherwise how do we deal with Integer query ids?
But not sure if using `<K>` would eliminate the warning.